File: graphedu/api/services/education/learning_event.py
"""学习事件上报 API 控制器

提供学生端通用事件上报接口。

主要接口：
- POST /education/learning-event  上报学习事件
"""

from fastapi import APIRouter, Body, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from graphedu.common.models.bo import CurrentUser
from graphedu.common.models.dto.educationv2.event import LearningEventCreateDTO
from graphedu.common.models.vo.base import ResponseType, ResponseUtil
from graphedu.common.resource.deps import get_db
from graphedu.security.auth import SecurityService
from graphedu.services.education.learning_event import LearningEventService

learning_event_controller = APIRouter(
    prefix="/education/learning-event", dependencies=[Depends(SecurityService.get_current_user)]
)


# ============================================================================
# 上报学习事件
# ============================================================================
@learning_event_controller.post(
    "",
    dependencies=[Depends(SecurityService.get_current_user)],
    response_model=ResponseType[dict],
)
async def report_learning_event(
    data: LearningEventCreateDTO = Body(),
    query_db: AsyncSession = Depends(get_db),
    current_user: CurrentUser = Depends(SecurityService.get_current_user),
):
    """上报学习事件

    学生端通用事件上报，支持以下事件类型：
    - chapter_open: 打开章节
    - interest: 标记感兴趣
    - explain_request: 请求解释
    - map_click: 知识图谱点击
    - tool_map_query: 工具查询
    """
    data.student_id = current_user.detail.student_info.student_id
    event_id = await LearningEventService.report_event(query_db, data)
    return ResponseUtil.success(data={"eventId": event_id})

File: graphedu-ui/src/api/education/learning_event.ts
/**
 * 学习事件上报相关 API
 * 对应后端：graphedu/api/services/education/learning_event.py
 */
import request from '@/utils/request'
import type { ResponseType } from '@/types/api/common.ts'

/**
 * 学习事件上报请求参数（前端精简版，studentId 由后端从 JWT 注入）
 */
export interface LearningEventReportDTO {
  /** 课程ID */
  courseId: number
  /** 事件类型 */
  eventType: 'chapter_open' | 'interest' | 'explain_request' | 'map_click' | 'tool_map_query'
  /** 章节ID */
  chapterId?: number
  /** 知识点UUID */
  nodeUuid?: string
  /** 事件文本内容 */
  eventContent?: string
  /** 事件扩展数据 */
  eventPayload?: Record<string, unknown>
  /** 会话ID */
  sessionId?: number
  /** 事件持续时长（秒） */
  durationSeconds?: number
}

/**
 * 上报学习事件
 * POST /education/learning-event
 */
export function reportLearningEvent(data: LearningEventReportDTO): Promise<ResponseType<{ eventId: number }>> {
  return request({
    url: '/education/learning-event',
    method: 'post',
    data,
  })
}
